addons/mail/static/src/core/common/thread_model.js

```diff
@@ -590,6 +772,370 @@ export class Thread extends Record {
         }
         return this.store.Message.get(Math.max(...previousMessages.map((m) => m.id)));
     }
+
+    async leave() {
+        await this.store.env.services.orm.call("discuss.channel", "action_unfollow", [this.id]);
+        this.delete();
+        const thread = this.store.discuss.channels.threads[0]
+            ? this.store.discuss.channels.threads[0]
+            : this.store.discuss.inbox;
+        thread?.setAsDiscussThread();
```

Comment:
thread should always be defined as it fallbacks on inbox

Reply:
Probably now, there's probably plan to not show inbox if not handle in odoo